src/core/qgismobileapp.cpp

```diff
@@ -800,13 +800,57 @@ void QgisMobileapp::readProjectFile()
 
   emit loadProjectEnded();
 
+  QgsRectangle projectExtent = getProjectExtent( mProjectFilePath );
+  if ( !projectExtent.isEmpty() )
+    extent = projectExtent;
+
   if ( !extent.isEmpty() && extent.width() != 0.0 )
   {
     // Add a bit of buffer so elements don't touch the map edges
     emit setMapExtent( extent.buffered( extent.width() * 0.02 ) );
   }
 }
 
+QgsRectangle QgisMobileapp::getProjectExtent( const QString &path )
+{
+  QgsRectangle extent;
+
+  QFileInfo fi( path );
+  if ( fi.exists() )
+  {
+    QSettings settings;
+    if ( fi.size() == settings.value( QStringLiteral( "/qgis/projectExtent/%1/filesize" ).arg( path ), 0 ).toLongLong() )
+    {
+      QStringList parts = settings.value( QStringLiteral( "/qgis/projectExtent/%1/extent" ).arg( path ), QString() ).toString().split( '|' );
+      if ( parts.size() == 4 )
+      {
+        extent.setXMinimum( parts[0].toDouble() );
+        extent.setXMaximum( parts[1].toDouble() );
+        extent.setYMinimum( parts[2].toDouble() );
+        extent.setYMaximum( parts[3].toDouble() );
+      }
+    }
+  }
+
+  return extent;
+}
+
+void QgisMobileapp::saveProjectExtent( const QgsRectangle &extent )
+{
+  QFileInfo fi( mProjectFilePath );
+  if ( fi.exists() )
+  {
+    QSettings settings;
+    settings.beginGroup( QStringLiteral( "/qgis/projectExtent/%1" ).arg( mProjectFilePath ) );
+    settings.setValue( QStringLiteral( "filesize" ), fi.size() );
```

Comment:
```suggestion
    settings.setValue( QStringLiteral( "filesize" ), fi.size() );
    settings.setValue( QStringLiteral( "mtime" ), fi.fileTime( QFileDevice::FileModificationTime ).toSecsSinceEpoch() );
```

Should we add that to the mix?

Reply:
@m-kuhn , well, I was thinking about it at first, but I feel it's a bit of an overkill maybe? Alllsoooo, I think it's bad because you might want to overwrite a project directory with updated data but not updated project file. The overwriting would change the modification file but it would still be the same project.

Hence filesize only.